File: Examples/ProgrammersGuide/WorkingWithGmailClient/WorkingWithGmailContacts/DeleteContact/VisualBasic/GoogleTestUser.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Text

Namespace Aspose.Email.Tests
	Public Class GoogleTestUser
		Inherits TestUser
		Public Sub New(ByVal name As String, ByVal eMail As String, ByVal password As String)
			Me.New(name, eMail, password, Nothing, Nothing, Nothing)
		End Sub

'INSTANT VB NOTE: The parameter clientId was renamed since Visual Basic will not uniquely identify class members when parameters have the same name:
'INSTANT VB NOTE: The parameter clientSecret was renamed since Visual Basic will not uniquely identify class members when parameters have the same name:
		Public Sub New(ByVal name As String, ByVal eMail As String, ByVal password As String, ByVal clientId_Renamed As String, ByVal clientSecret_Renamed As String)
			Me.New(name, eMail, password, clientId_Renamed, clientSecret_Renamed, Nothing)
		End Sub

'INSTANT VB NOTE: The parameter clientId was renamed since Visual Basic will not uniquely identify class members when parameters have the same name:
'INSTANT VB NOTE: The parameter clientSecret was renamed since Visual Basic will not uniquely identify class members when parameters have the same name:
'INSTANT VB NOTE: The parameter refreshToken was renamed since Visual Basic will not uniquely identify class members when parameters have the same name:
		Public Sub New(ByVal name As String, ByVal eMail As String, ByVal password As String, ByVal clientId_Renamed As String, ByVal clientSecret_Renamed As String, ByVal refreshToken_Renamed As String)
			MyBase.New(name, eMail, password, "gmail.com")
			ClientId = clientId_Renamed
			ClientSecret = clientSecret_Renamed
			RefreshToken = refreshToken_Renamed
		End Sub

		Public ReadOnly ClientId As String
		Public ReadOnly ClientSecret As String
		Public ReadOnly RefreshToken As String
	End Class
End Namespace

File: Examples/ProgrammersGuide/WorkingWithGmailClient/WorkingWithGmailContacts/DeleteContact/CSharp/GoogleTestUser.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Aspose.Email.Tests
{
    public class GoogleTestUser : TestUser
    {
        public GoogleTestUser(
            string name,
            string eMail,
            string password)
            : this(
                name,
                eMail,
                password,
                null,
                null,
                null)
        {
        }

        public GoogleTestUser(
            string name,
            string eMail,
            string password,
            string clientId,
            string clientSecret)
            : this(
                name,
                eMail,
                password,
                clientId,
                clientSecret,
                null)
        {
        }

        public GoogleTestUser(
            string name,
            string eMail,
            string password,
            string clientId,
            string clientSecret,
            string refreshToken)
            : base(
                name,
                eMail,
                password,
                "gmail.com")
        {
            ClientId = clientId;
            ClientSecret = clientSecret;
            RefreshToken = refreshToken;
        }

        public readonly string ClientId;
        public readonly string ClientSecret;
        public readonly string RefreshToken;
    }
}